tests/components/lacrosse_view/test_config_flow.py

```diff
@@ -161,3 +163,112 @@ async def test_form_unexpected_error(hass: HomeAssistant) -> None:
     assert result2["errors"] == {"base": "unknown"}
+
+
+async def test_already_configured_device(hass: HomeAssistant) -> None:
+    """Test we handle invalid auth."""
+    mock_config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        data={
+            "username": "test-username",
+            "password": "test-password",
+            "id": "1",
+            "name": "Test",
+        },
+        unique_id="1",
+    )
+    mock_config_entry.add_to_hass(hass)
+
+    # Now that we did the config once, let's try to do it again, this should raise the abort for already configured device
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == FlowResultType.FORM
+    assert result["errors"] is None
+
+    with patch("lacrosse_view.LaCrosse.login", return_value=True,), patch(
+        "lacrosse_view.LaCrosse.get_locations",
+        return_value=[Location(id=1, name="Test")],
+    ):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            {
+                "username": "test-username",
+                "password": "test-password",
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result2["type"] == FlowResultType.FORM
+    assert result2["step_id"] == "location"
+    assert result2["errors"] is None
+
+    with patch(
+        "homeassistant.components.lacrosse_view.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result3 = await hass.config_entries.flow.async_configure(
+            result2["flow_id"],
+            {
+                "location": "1",
+            },
+        )
+        await hass.async_block_till_done()
+
+    assert result3["type"] == FlowResultType.ABORT
+    assert result3["reason"] == "already_configured"
+    assert len(mock_setup_entry.mock_calls) == 0
+
+
+async def test_reauth(hass: HomeAssistant) -> None:
+    """Test reauthentication."""
+    data = {
+        "username": "test-username",
+        "password": "test-password",
+        "id": "1",
+        "name": "Test",
+    }
+    mock_config_entry = MockConfigEntry(
+        domain=DOMAIN,
+        data=data,
+        unique_id="1",
+        title="Test",
+    )
+    mock_config_entry.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_REAUTH}, data=data
+    )
+    assert result["step_id"] == "reauth_confirm"
+
+    result = await hass.config_entries.flow.async_configure(result["flow_id"], {})
+    assert result["type"] == FlowResultType.FORM
+    assert result["step_id"] == "user"
+
+    new_username = "new-username"
+    new_password = "new-password"
+
+    with patch("lacrosse_view.LaCrosse.login", return_value=True,), patch(
+        "lacrosse_view.LaCrosse.get_locations",
+        return_value=[Location(id=1, name="Test")],
```

Comment:
Please patch the integration setup function here too. The reauth flow will reload the config entry.